### PR TITLE
Controller events

### LINKED ISSUE #443: WDE should send 2 new controller events to capture the controller state - connected and disconnected
WDE should send 2 new controller events to capture the controller state - connected and disconnected

### PR DESCRIPTION
**Pull request by mary-dcouto**
close #443

### CODE REVIEW

File: service/ziti-tunnel/service/state.go [hunk 1]
```diff
@@ -301,6 +301,7 @@ func (t *RuntimeState) LoadIdentity(id *Id, refreshInterval int) {
 	}
 
 	id.CId = cziti.NewZid(sc)
+	id.CId.Fingerprint = id.FingerPrint
```
Comment (dovholuknf) [inline service/ziti-tunnel/service/state.go:304]: not sure this belongs here. why do you need to set this? - this should be coming from the csdk in the callback

File: service/cziti/sdk.go [hunk 2]
```diff
@@ -778,8 +778,23 @@ func zitiContextEvent(ztx C.ziti_context, status C.int, zid *ZIdentity) {
...
+		var controllerEvt = dto.ControllerEvent{
+			Fingerprint: zid.Fingerprint,
+			ControllerVersion: zid.Version,
+			IdentityName: zid.Name,
+			ActionEvent: dto.CONTROLLER_CONNECTED,
+		}
+		goapi.BroadcastEvent(controllerEvt)
 		log.Debugf("============ controller connected: %s at %v. MFA: %v", zid.Name, zid.Version, zid.MfaEnabled)
 	} else {
+		var controllerEvt = dto.ControllerEvent{
```
Comment (dovholuknf) [inline service/cziti/sdk.go:791]: imo you should move controllerEvt and the broadcast outside of this if just to reduce code and hopefully avoid any bugs if this code ever changes. just assign the ActionEvent to a var first, and do the create/broadcast after the if

File: service/ziti-tunnel/dto/dtos.go [hunk 3]
```diff
@@ -200,3 +200,10 @@ type MfaChallenge struct {
 
 type MfaResponse struct {
 }
+
+type ControllerEvent struct {
+	ActionEvent
+	Fingerprint 		string
+	ControllerVersion	string
+	IdentityName		string
```
Comment (dovholuknf) [inline service/ziti-tunnel/dto/dtos.go:208]: IdentityName is not needed in this struct nor is the ControllerVersion. This should just be Fingerprint

**Review by dovholuknf — APPROVED** [review]
make a few changes and it's gtg

File: service/ziti-tunnel/service/state.go (hunk 1 above)
Comment (mary-dcouto) [reply service/ziti-tunnel/service/state.go:304]: I dont think c-sdk maintains the fingerprint. This field is set in the zid.StatusChanges(int(status)) function, and this is called only after the controller event is received. I will remove the assignment, move the broadcast events operation after the StatusChanges function call.

Comment (dovholuknf) [reply service/ziti-tunnel/service/state.go:304]: when the identity is loaded either from cache or from the c sdk the fingerprint will be set. This is the wrong place to do it. see https://github.com/openziti/desktop-edge-win/blob/ca7b21447164ad99d7f9ed9a93948e490aa59fd1/service/ziti-tunnel/service/state.go#L273

Comment (mary-dcouto) [reply service/ziti-tunnel/service/state.go]: I just removed the fingerprint assignment that I added

File: service/ziti-tunnel/dto/dtos.go (hunk 3 above)
Comment (mary-dcouto) [reply service/ziti-tunnel/dto/dtos.go]: removed the unnecessary fields

**Pull request merged**